packs/signoz/templates/signoz-app.nomad.tpl

```diff
@@ -0,0 +1,149 @@
+# SigNoz Main Application Job
+job "[[ var "job_name" . ]]_signoz"  {
+  [[ template "region" . ]]
+  datacenters = [[ var "datacenters" . | toStringList ]]
+  type = "service"
+  node_pool   = [[ var "node_pool" . | quote ]]
+  group "signoz" {
+    network {
+      mode = "bridge"
+      port "http" { static = [[ var "signoz_http_port" . ]] }
+      port "http-internal" { static = [[ var "signoz_internal_port" . ]] }
+      port "opamp-internal" { static = [[ var "signoz_opamp_port" . ]] }
+    }
+
+    volume "signoz-db" {
+      type = "host"
+      access_mode = "single-node-single-writer"
+      source = [[ var "signoz_volume_name" . | quote ]]
+      attachment_mode="file-system"
+    }
+    # SigNoz initialization task
+    task "signoz-init" {
+      driver = "docker"
+      lifecycle {
+        hook = "prestart"
+        sidecar = false
+      }
+      
+      # Environment variables for ZooKeeper and ClickHouse
+      template {
+        env = true
+        data = <<EOH
+        {{range service "clickhouse-http"}}
+        CLICKHOUSE_PORT={{ .Port }}
+        CLICKHOUSE_HOST={{ .Address }}
+        {{end}}
+        EOH
+        destination = "secrets/hosts.env"
+        change_mode   = "restart"
+      }
+      
+      config {
+        image = "docker.io/busybox:1.35"
+        command = "sh"
+        args = [
+          "-c",
+          <<-EOT
+          echo "Waiting for ClickHouse HTTP ping..."
+          until wget -q --spider "http://$${CLICKHOUSE_HOST}:$${CLICKHOUSE_PORT}/ping"; do
+          echo "waiting for clickhouseDB (HTTP $${CLICKHOUSE_HOST}:$${CLICKHOUSE_PORT}/ping)"; sleep 5;
+          done
+          echo "ClickHouse HTTP is up"
+          EOT
+        ]
+      }
+    }
+
+    # Main SigNoz task
+    task "signoz" {
+      driver = "docker"
+
+      template {
+        env = true
+        data = <<EOH
+        {{range service "clickhouse-tcp"}}
+        CLICKHOUSE_PORT={{ .Port }}
+        CLICKHOUSE_HOST={{ .Address }}
+        {{end}}
+        EOH
+        destination = "secrets/clickhouse.env"
+        change_mode   = "restart"
+      }
+      template {
+        destination = "${NOMAD_SECRETS_DIR}/env.vars"
+        env         = true
+        change_mode = "restart"
+        data        = <<EOF
+{{- with nomadVar "nomad/jobs/[[ var "job_name" . ]]" -}}
```

Comment:
This needs to match with the actual job name:

```suggestion
{{- with nomadVar "nomad/jobs/[[ var "job_name" . ]]_signoz" -}}
```

But maybe "job name" is and incorrectly named variable if it's not actually the job name?``

Reply:
Rectified the var to be under the "nomad/jobs" so that it can accessible for all of the jobs under the signoz pack

Reply:
You didn't push that change, but that would also make the password available to _all_ jobs in the namespace, which we probably don't want. See https://developer.hashicorp.com/nomad/docs/concepts/variables#task-access-to-variables for details